torch/csrc/autograd/functions/accumulate_grad.h

```diff
@@ -50,6 +50,15 @@ struct TORCH_API AccumulateGrad : public Node {
     //     to all other Nodes). So we must lazily read the Tensor hooks here.
     return impl::hooks(variable);
   }
+
+  std::vector<std::unique_ptr<PostAccumulateGradHook>>&
+  tensor_post_acc_grad_hooks() noexcept {
+    // NB: Since the AccumulateGrad Node is only a weak ref from the Tensor,
+    //     it can be destroyed even though the Tensor is still alive (contrary
+    //     to all other Nodes). So we must lazily read the Tensor hooks here.
+    return impl::post_acc_grad_hooks(variable);
+  }
```

Comment:
Technically you don't even need this, just calling impl::post_acc_grad_hooks(variable) instead ::apply should be okay. The reason tensor_pre_hooks is separate out as a method here is so that it can override.

Reply:
OK I take this back given that we want to explicitly error w/ compiled_autograd